<launch>
<arg name="robot_namespace" default="/hauler_1"/>

  <!--TODO: load the bin joint commander -->
  <!--node name="cmd_joints" 
	pkg="motion_control" 
	type="cmd_joints"
	respawn="false" 
	output="screen" 
	ns="$(arg robot_namespace)"/>
    	<remap from="bin_joint_controller/command" to="$(arg robot_namespace)/bin_joint_controller/command"/-->

  <!-- load the lidar/camera mount joint commander -->
  <node name="cmd_sensor" 
	pkg="motion_control" 
	type="cmd_sensor"
	respawn="false" 
	output="screen" 
	ns="$(arg robot_namespace)"/>
    	<remap from="sensor_controller/command" to="$(arg robot_namespace)/sensor_controller/command"/>


  <!-- load the wheels' motor torques commander -->
  <node name="cmd_motors" 
	pkg="motion_control" 
	type="cmd_motors"
	respawn="false" 
	output="screen" 
	ns="$(arg robot_namespace)"/>
    	<remap from="fr_wheel_controller/command" to="$(arg robot_namespace)/fr_wheel_controller/command"/>
	<remap from="br_wheel_controller/command" to="$(arg robot_namespace)/br_wheel_controller/command"/>
    	<remap from="fl_wheel_controller/command" to="$(arg robot_namespace)/fl_wheel_controller/command"/>
	<remap from="bl_wheel_controller/command" to="$(arg robot_namespace)/bl_wheel_controller/command"/>


  <!-- load the steering arms' joints commander -->
  <node name="cmd_steering" 
	pkg="motion_control" 
	type="cmd_steering"
	respawn="false" 
	output="screen" 
	ns="$(arg robot_namespace)"/>
        <remap from="fr_steering_arm_controller/command" to="$(arg robot_namespace)/fr_steering_arm_controller/command"/>
	<remap from="br_steering_arm_controller/command" to="$(arg robot_namespace)/br_steering_arm_controller/command"/>
    	<remap from="fl_steering_arm_controller/command" to="$(arg robot_namespace)/fl_steering_arm_controller/command"/>
	<remap from="bl_steering_arm_controller/command" to="$(arg robot_namespace)/bl_steering_arm_controller/command"/>
        
	<remap from="/toggle_light" to="$(arg robot_namespace)/toggle_light"/>  
  <node name="teleop_modes_key" 
	pkg="motion_control" 
	type="teleop_modes_key"
	respawn="false" 
	output="screen" 
	ns="$(arg robot_namespace)"/>

</launch>




